ios/Exponent/Kernel/AppLoader/EXAppLoaderExpoUpdates.m

```diff
@@ -537,6 +538,19 @@ - (nullable EXManifestsManifest *)_processManifest:(EXManifestsManifest *)manife
     if (![mutableManifest[@"isVerified"] boolValue] && (EXEnvironment.sharedEnvironment.isManifestVerificationBypassed || [EXAppLoaderExpoUpdates _isAnonymousExperience:manifest])) {
       mutableManifest[@"isVerified"] = @(YES);
     }
+    
+    // when the manifest is not verified at this point, make the scope key and id a random value
+    if (![mutableManifest[@"isVerified"] boolValue]) {
+      NSString *randomValue = [[NSUUID UUID] UUIDString];
```

Comment:
Instead of a random value, I wonder if we should deterministically generate the actual scope key from the manifest's purported scope key. In other words, unverified projects would have isolation (each unverified project would have its own scope, consistent across reloads) without security (anyone can use any unverified scope key).

The motivation is to keep some of the existing behavior -- namely that unverified projects can be reloaded and will retain access to their own scoped data -- and to add security just to verified projects.

Reply:
> I wonder if we should deterministically generate the actual scope key from the manifest's purported scope key

Yeah, we could hash the purported scope key and prepend something like `sk-` (for debugging purposes as mentioned above).

> anyone can use any unverified scope key

The problem with this is that let's say we only prepended `sk-` to the purported scope key instead of also hashing it. Then, if an experience had a real scope key of `sk-real`, then the unverified app would just need to spoof their scope key to `real` to snoop. That being said, android actually allows this by just substituting out the modules. The hash on iOS could make it impossible to find the collision (I think).

Reply:
Yes, we would need to make sure that unverified scope keys are "namespaced" separately from verified scope keys. Hashing would do that as would just prefixing them with `unverified-` or any other way to make sure they can't spoof a verified scope key.

Reply:
question about the concrete values of `scopeKey`: is `scopeKey` still the same as the old `experienceId`? (`@username/slug`). do we expect this to change?

Reply:
`scopeKey` is currently the original `@account/slug` of the project when it was first created. If a project is renamed or transferred to another account, the scope key deliberately does not change (it has to stay stable). We made plans to start generating the scope keys for newly created projects to be random, opaque strings like `sk-tGXR8_Z5jdHi6B` to make clear that scope keys are unique, opaque values and use a string that doesn't contain any info that could change over time (like owner name or slug).

Reply:
> Hashing would do that as would just prefixing them with `unverified-` or any other way to make sure they can't spoof a verified scope key.

I think I'll go with a hash to ensure the namespace being separate (one-way-predictable and not reverse). That way, if there for some reason does happen to be a real scope key that just happens to have the prefix `unverfified-` it can't be spoofed.

Reply:
IMO the best solution is to do both: include a hash to avoid accidents, and prefix with something like `unverified-` to make the scope keys visibly unverified.

Reply:
Yep, went with salt (with the `unverified-` prefix) and hash, and then prefix with `unverified-<claimed-scope-key>` for debugging purposes.

So `@test/world` will become `unverified-@test/world-5a317b7d...`